jdr/jboss-as-jdr/src/main/java/org/jboss/as/jdr/util/JdrZipFile.java

```diff
@@ -135,8 +135,15 @@ public void add(VirtualFile file, InputStream is) throws Exception {
      * @throws Exception
      */
     public void add(String content, String path) throws Exception {
-        String name = "sos_strings/as7/" + path;
-        this.add(new ByteArrayInputStream(content.getBytes()), name);
+        StringBuilder name = new StringBuilder("sos_strings/");
+
+        name.append(this.env.getProductName().replace(" ", "_").toLowerCase());
+        name.append("-");
+        name.append(this.env.getProductVersion().split("\\.")[0]);
```

Comment:
This should never be null since they are set based on the management command response, `JdrRunner#obtainProductName()` and `JdrRunner#obtainProductVersion()` above. In case a product name or product version don't exist, management returns `undefined`. You can test this by running this patch against `build/` instead of `dist/`.